src/main/java/Duke.java

```diff
@@ -8,81 +12,58 @@
     public static List<TaskDTO> taskList = new ArrayList<>();
     public static int taskId = 0;
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
 
-        System.out.println("____________________________________________________________");
-        System.out.println("Hello! I'm Duke\n" +
-                "What can I do for you?");
 
         Scanner scanner = new Scanner(System.in);
 
-        processCommand(scanner);
-
-    }
+        UI ui = new UI();
+        UI.printWelcomeMessage();
+        String command = ui.readCommand();
 
-    public static void processCommand (Scanner scanner){
-        String command = scanner.nextLine();
-        TaskDTO newTask = new TaskDTO();
-        System.out.println("____________________________________________________________");
-        if (command.equalsIgnoreCase("list")){
-            StringBuilder output = new StringBuilder("");
+        while (!ParserUtil.getFunctionCommand(command).equals("bye")) {
+            processCommand(command);
+            command = ui.readCommand();
+        }
 
-            output.append(" Here are the tasks in your list:\n");
-            for (TaskDTO task : taskList){
+    }
 
-                output.append(task.toString());
 
-            }
-            System.out.println(output.toString());
-        }else if(command.toLowerCase().startsWith("done ")){
-            String[] spitList = command.split("done ");
-            for (TaskDTO task:taskList){
-                if (Integer.parseInt(spitList[1]) == task.getTaskId()){
-                    task.processDoneTask();
+    public static void processCommand(String originalCommand) {
+        System.out.println("____________________________________________________________");
+        String operation = ParserUtil.getFunctionCommand(originalCommand);
+        String details = originalCommand.substring(originalCommand.indexOf(' ') + 1);
+
+
+        switch (operation){
+            case "todo":
```

Comment:
Perhaps we can align the indentation for switch and case as there should be no indentation for case clauses.